pennylane/beta/devices/default_tensor.py

```diff
@@ -351,7 +351,8 @@ def _add_state_prep_nodes(self, operation, par):
 
         elif operation == "BasisState":
             n = len(par[0])
-            if n == 0 or n > self.num_wires or not set(par[0]).issubset({0, 1}):
+            elements = set(par[0].tolist())
```

Comment:
Since indexing into a `qml.numpy` array now always returns an array, in several places we now need to explicitly cast to lists if we do _not_ want the output to remain an array.